### PR TITLE
Create check run through app with fallback for when app is not installed

### PR DESCRIPTION
**Pull request by ajinkya599**
- Create check run through app proxy.
- Fallback to check run creation using GitHub Actions app when app is not installed.

### CODE REVIEW

File: src/githubClient.ts [hunk 1]
```diff
@@ -6,7 +8,35 @@ export class GitHubClient {
         this._token = token;
     }
 
-    public async createCheckRun(payload: any): Promise<void> {
+    public async createCheckRunWithFallback(checkRunThroughAppPayload: any, checkRunPayload: any): Promise<void> {
```
Comment (ds-ms) [inline src/githubClient.ts:11]: Shouldn't "Fallback" be implicit / default expectation?

File: src/githubClient.ts [hunk 2]
```diff
@@ -6,7 +8,35 @@ export class GitHubClient {
...
+        const checkRunUrl = `https://api.github.com/repos/${this._repository}/container-scanning/check-run`;
+        const webRequest = new WebRequest();
+        webRequest.method = "POST";
+        webRequest.uri = checkRunUrl;
+        webRequest.body = JSON.stringify(checkRunThroughAppPayload);
+        webRequest.headers = {
+            Authorization: `Bearer ${this._token}`
+        };
+
+        console.log(`Creating check run. image_name: ${checkRunThroughAppPayload['image_name']}, head_sha: ${checkRunThroughAppPayload['head_sha']}`);
```
Comment (ds-ms) [inline src/githubClient.ts:21]: call it scan-result payload maybe?

Comment (ds-ms) [reply src/githubClient.ts:21]: or just payload

File: src/githubClient.ts [hunk 3]
```diff
@@ -6,7 +8,35 @@ export class GitHubClient {
...
+        const response: WebResponse = await sendRequest(webRequest);
+        core.debug(util.format('Response from scanitizer app:\n', response.body));
+        if (response.statusCode == StatusCodes.UNPROCESSABLE_ENTITY
+            && response.body
+            && response.body.message
+            && response.body.message.error_code === 'APP_NOT_INSTALLED') {
+            console.log('Looks like the scanitizer app is not installed on the repo. Falling back to check run creation through GitHub actions app...');
```
Comment (ds-ms) [inline src/githubClient.ts:28]: Get this reviewed by PM. Also, we need to figure out a different entry point for this. This won't be discoverable in a plethora of logs!

File: src/main.ts [hunk 4]
```diff
@@ -124,9 +124,10 @@ async function run(): Promise<void> {
     }
 
     try {
-        const checkRunPayload = utils.getCheckRunPayloadWithScanResult(trivyStatus, dockleStatus);
+        const checkRunPayload = utils.getCheckRunPayload(trivyStatus, dockleStatus);
+        const checkRunThroughAppPayload = utils.getCheckRunThroughAppPayload(trivyStatus, dockleStatus);
         const githubClient = new GitHubClient(process.env.GITHUB_REPOSITORY, inputHelper.githubToken);
-        await githubClient.createCheckRun(checkRunPayload);
+        await githubClient.createCheckRunWithFallback(checkRunThroughAppPayload, checkRunPayload);
```
Comment (ds-ms) [inline src/main.ts:130]: It's not the client's duty to worry about fall back ideally. The validation of status logic should be here.

File: src/githubClient.ts (hunk 3 above)
Comment (ds-ms) [reply src/githubClient.ts:28]: cc: @raiyanalam

File: src/githubClient.ts (hunk 1 above)
Comment (ajinkya599) [reply src/githubClient.ts]: Refactored as discussed offline.

File: src/main.ts (hunk 4 above)
Comment (ajinkya599) [reply src/main.ts]: Moved the logic to utils.

File: src/githubClient.ts (hunk 2 above)
Comment (ajinkya599) [reply src/githubClient.ts]: changed to payload.

File: src/githubClient.ts (hunk 3 above)
Comment (ds-ms) [reply src/githubClient.ts]: cc: @azooinmyluggage

Comment (ajinkya599) [reply src/githubClient.ts]: Filed an issue to track this:
https://github.com/Azure/container-scan/issues/22

**Pull request merged**